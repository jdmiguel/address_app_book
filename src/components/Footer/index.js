import React from 'react';

import { literals } from '../../utils/constants';

const {
  footer: { text, textLink, urlLink },
} = literals;

const Footer = () => (
  <footer>
    <div />
    <div className="container">
      <p>
        <a href={urlLink}>{textLink}</a>
        {text}
      </p>
    </div>
  </footer>
);

export default Footer;
