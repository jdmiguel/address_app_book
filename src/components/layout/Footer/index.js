import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import { literals, settingsIcon, homeIcon } from '../../../utils/constants';

const {
  footer: {
    text,
    link: { settingsText, settingsLink, homeText, homeLink },
    warning,
  },
} = literals;

const Footer = ({ withWarning, withSettings }) => (
  <footer className={withWarning ? 'withWarning' : ''}>
    <div>{withWarning && warning}</div>
    <div className="container">
      <Link to={withSettings ? settingsLink : homeLink}>
        <img src={withSettings ? settingsIcon : homeIcon} alt="icon" />
        {withSettings ? settingsText : homeText}
      </Link>
      {text}
    </div>
  </footer>
);

Footer.propTypes = {
  withWarning: PropTypes.bool,
  withSettings: PropTypes.bool,
};

Footer.defaultProps = {
  withWarning: false,
  withSettings: false,
};

export default Footer;
