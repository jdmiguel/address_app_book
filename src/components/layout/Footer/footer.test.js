import React from 'react';
import { render } from '@testing-library/react';

import Footer from '.';

describe('Component: Footer', () => {
  it('should render with home link', () => {
    const { container } = render(<Footer />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with settings link', () => {
    const { container } = render(<Footer withSettings />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with home link and warning msg', () => {
    const { container } = render(<Footer withWarning />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render with settings link and warning msg', () => {
    const { container } = render(<Footer withSettings withWarning />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
