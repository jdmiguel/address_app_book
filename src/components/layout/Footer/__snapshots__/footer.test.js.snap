// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Footer should render with home link 1`] = `
<footer>
  <div
    class="container"
  >
    <a
      href="/"
    >
      <img
        alt="home"
        src="function svgrLoader(source) {
  const callback = this.async();

  const _ref = (0, _loaderUtils.getOptions)(this) || {},
        {
    babel = true
  } = _ref,
        options = _objectWithoutPropertiesLoose(_ref, [\\"babel\\"]);

  const readSvg = () => new Promise((resolve, reject) => {
    this.fs.readFile(this.resourcePath, (err, result) => {
      if (err) reject(err);
      resolve(result);
    });
  });

  const previousExport = (() => {
    if (source.toString('utf-8').startsWith('export ')) {
      return source;
    }

    const exportMatches = source.toString('utf-8').match(/^module.exports\\\\s*=\\\\s*(.*)/);
    return exportMatches ? \`export default \${exportMatches[1]}\` : null;
  })();

  const tranformSvg = svg => (0, _core2.default)(svg, options, {
    caller: {
      name: '@svgr/webpack',
      previousExport,
      defaultPlugins: [_pluginSvgo.default, _pluginJsx.default]
    },
    filePath: this.resourcePath
  }).then(jsCode => {
    if (!babel) return jsCode;
    return (0, _core.transformAsync)(jsCode, babelOptions).then(({
      code
    }) => code);
  }).then(result => callback(null, result)).catch(err => callback(err));

  if (previousExport) {
    readSvg().then(tranformSvg);
  } else {
    tranformSvg(source);
  }
}"
      />
      Back to Home
    </a>
    This is a coding exercise for Sherpany company
  </div>
</footer>
`;

exports[`Component: Footer should render with home link and warning msg 1`] = `
<footer>
  <div
    class="warning"
  >
    <img
      alt="warning"
      src="function svgrLoader(source) {
  const callback = this.async();

  const _ref = (0, _loaderUtils.getOptions)(this) || {},
        {
    babel = true
  } = _ref,
        options = _objectWithoutPropertiesLoose(_ref, [\\"babel\\"]);

  const readSvg = () => new Promise((resolve, reject) => {
    this.fs.readFile(this.resourcePath, (err, result) => {
      if (err) reject(err);
      resolve(result);
    });
  });

  const previousExport = (() => {
    if (source.toString('utf-8').startsWith('export ')) {
      return source;
    }

    const exportMatches = source.toString('utf-8').match(/^module.exports\\\\s*=\\\\s*(.*)/);
    return exportMatches ? \`export default \${exportMatches[1]}\` : null;
  })();

  const tranformSvg = svg => (0, _core2.default)(svg, options, {
    caller: {
      name: '@svgr/webpack',
      previousExport,
      defaultPlugins: [_pluginSvgo.default, _pluginJsx.default]
    },
    filePath: this.resourcePath
  }).then(jsCode => {
    if (!babel) return jsCode;
    return (0, _core.transformAsync)(jsCode, babelOptions).then(({
      code
    }) => code);
  }).then(result => callback(null, result)).catch(err => callback(err));

  if (previousExport) {
    readSvg().then(tranformSvg);
  } else {
    tranformSvg(source);
  }
}"
    />
    While finder is active, more users will not be loaded by scrolling
  </div>
  <div
    class="container"
  >
    <a
      href="/"
    >
      <img
        alt="home"
        src="function svgrLoader(source) {
  const callback = this.async();

  const _ref = (0, _loaderUtils.getOptions)(this) || {},
        {
    babel = true
  } = _ref,
        options = _objectWithoutPropertiesLoose(_ref, [\\"babel\\"]);

  const readSvg = () => new Promise((resolve, reject) => {
    this.fs.readFile(this.resourcePath, (err, result) => {
      if (err) reject(err);
      resolve(result);
    });
  });

  const previousExport = (() => {
    if (source.toString('utf-8').startsWith('export ')) {
      return source;
    }

    const exportMatches = source.toString('utf-8').match(/^module.exports\\\\s*=\\\\s*(.*)/);
    return exportMatches ? \`export default \${exportMatches[1]}\` : null;
  })();

  const tranformSvg = svg => (0, _core2.default)(svg, options, {
    caller: {
      name: '@svgr/webpack',
      previousExport,
      defaultPlugins: [_pluginSvgo.default, _pluginJsx.default]
    },
    filePath: this.resourcePath
  }).then(jsCode => {
    if (!babel) return jsCode;
    return (0, _core.transformAsync)(jsCode, babelOptions).then(({
      code
    }) => code);
  }).then(result => callback(null, result)).catch(err => callback(err));

  if (previousExport) {
    readSvg().then(tranformSvg);
  } else {
    tranformSvg(source);
  }
}"
      />
      Back to Home
    </a>
    This is a coding exercise for Sherpany company
  </div>
</footer>
`;

exports[`Component: Footer should render with settings link 1`] = `
<footer>
  <div
    class="container"
  >
    <a
      href="/settings"
    >
      <img
        alt="settings"
        src="function svgrLoader(source) {
  const callback = this.async();

  const _ref = (0, _loaderUtils.getOptions)(this) || {},
        {
    babel = true
  } = _ref,
        options = _objectWithoutPropertiesLoose(_ref, [\\"babel\\"]);

  const readSvg = () => new Promise((resolve, reject) => {
    this.fs.readFile(this.resourcePath, (err, result) => {
      if (err) reject(err);
      resolve(result);
    });
  });

  const previousExport = (() => {
    if (source.toString('utf-8').startsWith('export ')) {
      return source;
    }

    const exportMatches = source.toString('utf-8').match(/^module.exports\\\\s*=\\\\s*(.*)/);
    return exportMatches ? \`export default \${exportMatches[1]}\` : null;
  })();

  const tranformSvg = svg => (0, _core2.default)(svg, options, {
    caller: {
      name: '@svgr/webpack',
      previousExport,
      defaultPlugins: [_pluginSvgo.default, _pluginJsx.default]
    },
    filePath: this.resourcePath
  }).then(jsCode => {
    if (!babel) return jsCode;
    return (0, _core.transformAsync)(jsCode, babelOptions).then(({
      code
    }) => code);
  }).then(result => callback(null, result)).catch(err => callback(err));

  if (previousExport) {
    readSvg().then(tranformSvg);
  } else {
    tranformSvg(source);
  }
}"
      />
      Go to Settings
    </a>
    This is a coding exercise for Sherpany company
  </div>
</footer>
`;

exports[`Component: Footer should render with settings link and warning msg 1`] = `
<footer>
  <div
    class="warning"
  >
    <img
      alt="warning"
      src="function svgrLoader(source) {
  const callback = this.async();

  const _ref = (0, _loaderUtils.getOptions)(this) || {},
        {
    babel = true
  } = _ref,
        options = _objectWithoutPropertiesLoose(_ref, [\\"babel\\"]);

  const readSvg = () => new Promise((resolve, reject) => {
    this.fs.readFile(this.resourcePath, (err, result) => {
      if (err) reject(err);
      resolve(result);
    });
  });

  const previousExport = (() => {
    if (source.toString('utf-8').startsWith('export ')) {
      return source;
    }

    const exportMatches = source.toString('utf-8').match(/^module.exports\\\\s*=\\\\s*(.*)/);
    return exportMatches ? \`export default \${exportMatches[1]}\` : null;
  })();

  const tranformSvg = svg => (0, _core2.default)(svg, options, {
    caller: {
      name: '@svgr/webpack',
      previousExport,
      defaultPlugins: [_pluginSvgo.default, _pluginJsx.default]
    },
    filePath: this.resourcePath
  }).then(jsCode => {
    if (!babel) return jsCode;
    return (0, _core.transformAsync)(jsCode, babelOptions).then(({
      code
    }) => code);
  }).then(result => callback(null, result)).catch(err => callback(err));

  if (previousExport) {
    readSvg().then(tranformSvg);
  } else {
    tranformSvg(source);
  }
}"
    />
    While finder is active, more users will not be loaded by scrolling
  </div>
  <div
    class="container"
  >
    <a
      href="/settings"
    >
      <img
        alt="settings"
        src="function svgrLoader(source) {
  const callback = this.async();

  const _ref = (0, _loaderUtils.getOptions)(this) || {},
        {
    babel = true
  } = _ref,
        options = _objectWithoutPropertiesLoose(_ref, [\\"babel\\"]);

  const readSvg = () => new Promise((resolve, reject) => {
    this.fs.readFile(this.resourcePath, (err, result) => {
      if (err) reject(err);
      resolve(result);
    });
  });

  const previousExport = (() => {
    if (source.toString('utf-8').startsWith('export ')) {
      return source;
    }

    const exportMatches = source.toString('utf-8').match(/^module.exports\\\\s*=\\\\s*(.*)/);
    return exportMatches ? \`export default \${exportMatches[1]}\` : null;
  })();

  const tranformSvg = svg => (0, _core2.default)(svg, options, {
    caller: {
      name: '@svgr/webpack',
      previousExport,
      defaultPlugins: [_pluginSvgo.default, _pluginJsx.default]
    },
    filePath: this.resourcePath
  }).then(jsCode => {
    if (!babel) return jsCode;
    return (0, _core.transformAsync)(jsCode, babelOptions).then(({
      code
    }) => code);
  }).then(result => callback(null, result)).catch(err => callback(err));

  if (previousExport) {
    readSvg().then(tranformSvg);
  } else {
    tranformSvg(source);
  }
}"
      />
      Go to Settings
    </a>
    This is a coding exercise for Sherpany company
  </div>
</footer>
`;
